vos/vos/storage_inventory.py

```diff
@@ -0,0 +1,315 @@
+        elif not destination:
+            raise ValueError('Destination is mandatory.')
+
+        logger.debug('Checking sanity of inputs.')
+        is_source_remote = self._is_remote(source)
+        is_destination_remote = self._is_remote(destination)
+
+        if is_source_remote and is_destination_remote:
+            raise ValueError('Unable to process server to server copying.')
+        elif is_source_remote:
+            logging.debug(
+                'Source is remote.  Downloading from {} to {}'.format(
+                    source, destination))
+            destination_size = self._get(source, destination)
+            if send_md5:
+                return md5_cache.MD5Cache.compute_md5(destination)
+            elif disposition:
+                # TODO: Should this return the name of the destination instead?
+                # TODO: jenkinsd 2020.02.05
+                #
+                return None
+            else:
+                return destination_size
+        elif is_destination_remote:
+            logging.debug(
+                'Destination is remote.  Uploading from {} to {}'.format(
+                    source, destination))
+            metadata = self._put(source, destination)
+            if send_md5:
+                return metadata.get('content_md5')
+            elif disposition:
+                return metadata.get('content_disposition')
+            else:
+                return metadata.get('content_length')
+        else:
+            raise ValueError('Unable to process copying {} to {}'.format(
+                                source, destination))
+
+    def delete(self, uri):
+        """Delete the Artifact
+        :param uri: The Artifact URI to delete.
+
+        :raises When a network problem occurs, it raises one of the
+        HttpException exceptions declared in the
+        cadcutils.exceptions module
+        """
+        logger.debug("delete {0}".format(uri))
+        url = '{}/{}'.format(
+            self._get_ws_client()._get_url((STANDARD_ID, None)), uri)
+        response = self.conn.session.delete(url)
+        response.raise_for_status()
+
+    def transfer(self, uri, direction, view=None, cutout=None):
+        transfer_url = '{}/{}'.format(
+            self._get_ws_client()._get_url((STANDARD_ID, None)), uri)
+        trans = Transfer()
+        return trans.transfer(transfer_url, uri, direction, self.conn,
+                              self._get_protocol(), view, cutout)
+
+    def get_transfer_error(self, uri, url):
+        trans = Transfer()
+        return trans.get_transfer_error(self.conn, uri, url)
+
+    def _get(self, source, destination):
+        """
+        :param source: The source Artifact to download (URI).
+        :type uri:  unicode
+        :param destination: The destination to write the data to (path).
+        :type destination: unicode
+        :return: destination size in bytes
+        :rtype long
+        """
```

Comment:
This, unlike vos, transfers bytes automatically even if source and destination are same?